frontend/src/components/WalletBalance.tsx

```diff
@@ -0,0 +1,35 @@
+import { BigNumber, BigNumberish } from 'ethers'
+import styled from 'styled-components/macro'
+import tw from 'twin.macro'
+import { formatAmount } from '../utils/amount'
+
+interface Props {
+  parsedAmount: BigNumber
+  walletAmount: BigNumber
+  decimals: number
+  symbol: string
+}
```

Comment:
Quick qst: would it be better to use the wallet context here instead of getting those as props?

Reply:
Both are good approaches imo. Went with passing in props for this one to generally make all the UI components as dumb as possible and leave the data handling to either the context or the parent component